modules/_user_data/main.tf

```diff
@@ -1,65 +1,49 @@
 
 locals {
-  int_linux_default_user_data = var.create && var.platform == "linux" && (var.enable_bootstrap_user_data || var.user_data_template_path != "") ? base64encode(templatefile(
-    coalesce(var.user_data_template_path, "${path.module}/../../templates/linux_user_data.tpl"),
+  template_path = {
+    al2023       = "${path.module}/../../templates/al2023_user_data.tpl"
+    bottlerocket = "${path.module}/../../templates/bottlerocket_user_data.tpl"
+    linux        = "${path.module}/../../templates/linux_user_data.tpl"
+    windows      = "${path.module}/../../templates/windows_user_data.tpl"
+  }
+
+  user_data = base64encode(templatefile(
+    coalesce(var.user_data_template_path, local.template_path[var.platform]),
     {
       # https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#launch-template-custom-ami
       enable_bootstrap_user_data = var.enable_bootstrap_user_data
+
       # Required to bootstrap node
       cluster_name        = var.cluster_name
       cluster_endpoint    = var.cluster_endpoint
       cluster_auth_base64 = var.cluster_auth_base64
+
       # Optional
-      cluster_service_ipv4_cidr = var.cluster_service_ipv4_cidr != null ? var.cluster_service_ipv4_cidr : ""
+      cluster_service_ipv4_cidr = var.cluster_service_ipv4_cidr != null && contains(["linux", "al2023"], var.platform) ? var.cluster_service_ipv4_cidr : ""
```

Comment:
Is this necessary to have `contains()` here? Normally, we ask users to specify values (`var.cluster_service_ipv4_cidr`) if they want, and they are responsible for whether it makes sense to be used there or not.

Reply:
after re-visiting this, its a redundant guard that is not necessary. The operating systems that do not support this, they don't have the variable in their respective templates so we can remove the entire `contains()` - removed and updated examples to ensure no-op https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2942/commits/202b43fb70b78e12c51fd598b709cacd7cce3081